packages/ckeditor5-engine/tests/conversion/upcasthelpers.js

```diff
@@ -355,6 +356,44 @@ describe( 'UpcastHelpers', () => {
 			);
 		} );
 
+		// https://github.com/ckeditor/ckeditor5/issues/10800
+		it( 'should consume element at a low-1 priority so that converters on lower priorities will not convert it', () => {
+			upcastHelpers.elementToAttribute( {
+				model: 'attribA',
+				view: { name: 'a', attributes: 'href' }
+			} );
+
+			upcastDispatcher.on( 'element:a', ( evt, data, conversionApi ) => {
+				if ( !conversionApi.consumable.consume( data.viewItem, { name: true } ) ) {
+					return;
+				}
+
+				conversionApi.writer.setAttribute( 'htmlA', 'true', data.modelRange );
+			}, { priority: priorities.get( 'low' ) - 1 } );
+
+			expectResult(
+				new ViewAttributeElement( viewDocument, 'a', { href: 'foo' }, new ViewText( viewDocument, 'foo' ) ),
+				'<$text attribA="true">foo</$text>'
+			);
+		} );
+
+		// https://github.com/ckeditor/ckeditor5/issues/10800
+		it( 'should convert children even if some custom converter consumed attributes without triggering children conversion', () => {
+			upcastHelpers.elementToAttribute( {
+				model: 'attribA',
+				view: { name: 'a', attributes: 'href' }
+			} );
+
+			upcastDispatcher.on( 'element:a', ( evt, data, conversionApi ) => {
+				conversionApi.consumable.consume( data.viewItem, { attributes: [ 'href' ] } );
+			}, { priority: 'high' } );
+
+			expectResult(
+				new ViewAttributeElement( viewDocument, 'a', { href: 'foo' }, new ViewText( viewDocument, 'foo' ) ),
+				'foo'
+			);
+		} );
```

Comment:
Unfortunately I'm not able to understand what this is testing exactly, because it's testing at the unit level — and pretty much almost at the _function_ level. I _would_ be able to understand this if it were testing `input markup → function calls → output markup` 🤓  But I understand that's not the right level to test or perhaps even too slow given the size of your test suite.

So instead of a thorough review, I'm asking a simple question: **does this test both the scenario in https://github.com/ckeditor/ckeditor5/issues/10800 and in https://github.com/ckeditor/ckeditor5/issues/10800#issuecomment-960654976**?